controller/hybridgateway/metadata/namespace.go

```diff
@@ -0,0 +1,16 @@
+package metadata
+
+import (
+	gwtypes "github.com/kong/kong-operator/internal/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NamespaceFromParentRef returns the namespace for a given Route ParentReference.
+// If the ParentReference specifies a namespace, that namespace is returned.
+// Otherwise, the namespace of the provided resource is returned.
+func NamespaceFromParentRef(obj metav1.Object, pRef *gwtypes.ParentReference) string {
+	if pRef.Namespace != nil && *pRef.Namespace != "" {
+		return string(*pRef.Namespace)
+	}
```

Comment:
```suggestion
	if ns := lo.FromPtr(pRef.Namespace); ns != "" {
		return string(ns)
	}
```
WDYT?

Furthermore, it is rather invalid to pass `pRef = nil` to this function. Are you checking it on the caller side?

Reply:
Good catch! While the call stack heading here in current code ensures that pRef will not be nil, the check in this function of the metadata package is mandatory IMHO! 👍🏼